tests/test_tree_kem_kat.rs

```diff
@@ -0,0 +1,278 @@
+            tests.push(test);
+        }
+    }
+
+    write("test_vectors/kat_tree_kem_openmls-new.json", &tests);
+}
+
+#[cfg(any(feature = "expose-test-vectors", test))]
+pub fn generate_test_vector(n_leaves: u32, ciphersuite: &'static Ciphersuite) -> TreeKemTestVector {
+    // The test really only makes sense with two or more leaves
+    if n_leaves <= 1 {
+        panic!("test vector can only be generated with two or more members")
+    }
+    // Set up a group with `n_leaves` members.
+    let handshake_message_format = HandshakeMessageFormat::Plaintext;
+    let update_policy = UpdatePolicy::default();
+    let callbacks = ManagedGroupCallbacks::default();
+    let managed_group_config =
+        ManagedGroupConfig::new(handshake_message_format, update_policy, 0, 0, callbacks);
+    let setup = ManagedTestSetup::new(managed_group_config, n_leaves as usize + 10);
+
+    // - I am the client with key package `my_key_package`
+    // - I was added by the client at leaf index add_sender
+    // - I therefore need to initialize my own path with the `path_secret` at the correct index
+    // - Then the client at `update_sender` sends an update with the given `update_path`
+    // - I process that update
+
+    // We now need to create a state, where a client is added to a random
+    // position in the tree by another randomly chosen client.
+
+    // To reach that state, we create a group of `n_leaves` members and remove a
+    // member from a random position.
+    let group_id = setup
+        .create_random_group(n_leaves as usize, ciphersuite)
+        .unwrap();
+
+    let mut groups = setup.groups.borrow_mut();
+    let group = groups.get_mut(&group_id).unwrap();
+    let remover_id = group.random_group_member();
+    let mut target_id = group.random_group_member();
+    while remover_id == target_id {
+        target_id = group.random_group_member();
+    }
+    //let remover_index = thread_rng().gen_range(0..n_leaves);
+    println!("remover id: {:?}", remover_id);
+    //let mut target_index = thread_rng().gen_range(0..n_leaves);
+    //while remover_index == target_index {
+    //    target_index = thread_rng().gen_range(0..n_leaves);
+    //}
+    println!("target id: {:?}", target_id);
+
+    let (target_index, _) = group
+        .members
+        .iter()
+        .find(|(_, id)| id == &target_id)
+        .unwrap()
+        .clone();
+
+    setup
+        .remove_clients_by_index(
+            ActionType::Commit,
+            group,
+            &remover_id,
+            &[target_index as usize],
+        )
+        .unwrap();
+
+    // We then have the same client who removed the target add a fresh member.
+    let adder_id = remover_id;
+    println!("adder id: {:?}", adder_id);
+    let (adder_index, _) = group
+        .members
+        .iter()
+        .find(|(_, id)| id == &adder_id)
+        .unwrap()
+        .clone();
+    let addees = setup.random_new_members_for_group(group, 1).unwrap();
```

Comment:
Instead of using this as `addees[0]` every time you could actually get out the one addee in there.